module-network.tf

```diff
@@ -0,0 +1,142 @@
+  state_id         = random_id.state_id.id
+  compartment_id   = coalesce(var.network_compartment_id, local.compartment_id)
+  defined_tags     = lookup(var.defined_tags, "network", {})
+  freeform_tags    = lookup(var.freeform_tags, "network", {})
+  tag_namespace    = var.tag_namespace
+  use_defined_tags = var.use_defined_tags
+
+  allow_node_port_access       = var.allow_node_port_access
+  allow_pod_internet_access    = var.allow_pod_internet_access
+  allow_rules_internal_lb      = var.allow_rules_internal_lb
+  allow_rules_public_lb        = var.allow_rules_public_lb
+  allow_worker_internet_access = var.allow_worker_internet_access
+  allow_worker_ssh_access      = var.allow_worker_ssh_access
+  assign_dns                   = var.assign_dns
+  bastion_allowed_cidrs        = var.bastion_allowed_cidrs
+  bastion_type                 = var.bastion_type
+  cni_type                     = var.cni_type
+  control_plane_allowed_cidrs  = var.control_plane_allowed_cidrs
+  control_plane_type           = var.control_plane_type
+  create_bastion               = var.create_bastion
+  create_fss                   = var.create_fss
+  create_nsgs                  = var.create_nsgs
+  create_operator              = local.operator_enabled
+  enable_waf                   = var.enable_waf
+  ig_route_table_id            = local.ig_route_table_id
+  load_balancers               = var.load_balancers
+  nat_route_table_id           = local.nat_route_table_id
+  subnets                      = var.subnets
+  vcn_cidrs                    = local.vcn_cidrs
+  vcn_id                       = local.vcn_id
+  worker_type                  = var.worker_type
+
+  providers = {
+    oci.home = oci.home
+  }
+}
+
+output "ig_route_table_id" {
+  description = "Internet gateway route table ID"
+  value       = local.ig_route_table_id
+}
+
+output "nat_route_table_id" {
+  description = "NAT gateway route table ID"
+  value       = local.nat_route_table_id
+}
+
+output "nsg_ids" {
+  description = "Map of network security group IDs by role for the cluster and associated resources."
+  value = var.create_nsgs ? {
+    "bastion"  = module.network.bastion_nsg_id
+    "cp"       = module.network.cp_nsg_id
+    "fss"      = module.network.fss_nsg_id
+    "int_lb"   = module.network.int_lb_nsg_id
+    "operator" = module.network.operator_nsg_id
+    "pods"     = module.network.pod_nsg_id
+    "pub_lb"   = module.network.pub_lb_nsg_id
+    "workers"  = module.network.worker_nsg_id
+  } : null
+}
+
+output "subnet_ids" {
+  description = "Map of subnet ids by role for the cluster and associated resources."
+  value       = module.network.subnet_ids
+}
+
+output "drg_id" {
+  description = "Dynamic routing gateway ID"
+  value       = one(module.drg[*].drg_id)
+}
+
+output "vcn_id" {
+  description = "VCN ID"
+  value       = local.vcn_id
+}
+
+output "network_security_rules" {
+  value = var.output_network_detail ? module.network.network_security_rules : null
```

Comment:
Let's have a `detailed_output` variable that covers everything. No need for a specific one for network.

If it's false, then we give a minimal output.